### PR TITLE
fix(upnp): use correct mapping protocol for UDP

### PR DESCRIPTION
**Pull request by arpankapoor**
## Description

Return `PortMappingProtocol::UDP` when the multiaddress protocol is UDP.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by jxs — APPROVED** [review]
LGTM, thanks for this!

**Review by thomaseizinger — DISMISSED** [review]

**Comment by thomaseizinger** [comment]
This needs a changelog entry and a version bump, otherwise we can release it!

**Review by jxs — DISMISSED** [review]

File: protocols/upnp/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.1.1 - unreleased
+
+- Fix port mapping protocol used for a UDP multiaddress
```
Comment (thomaseizinger) [inline protocols/upnp/CHANGELOG.md]:
```suggestion
- Fix port mapping protocol used for a UDP multiaddress.
  See [PR 4542](https://github.com/libp2p/rust-libp2p/pull/4542).
```

**Review by thomaseizinger — COMMENTED** [review]
You'll also need to bump the version in the root `Cargo.toml`, see CI error :)

**Comment by thomaseizinger** [comment]
Thank you!

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**